File: src/my_cpp_pkg/src/smartphone.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"
#include <string>

class SmartphoneNode : public rclcpp::Node
{
public:
    SmartphoneNode() : Node("smartphone")
    {
        __subscriber = this->create_subscription<example_interfaces::msg::String>(
            "robot_news", 10, std::bind(&SmartphoneNode::callbackRobotNews, this, std::placeholders::_1));
        RCLCPP_INFO(this->get_logger(), "C++ Smartphone node has started!");
    }

private:
    rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr __subscriber;

    void callbackRobotNews(const example_interfaces::msg::String msg)
    {
        RCLCPP_INFO(this->get_logger(), msg.data.c_str());
    }
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SmartphoneNode>(); // Cpp nodes are created as shared pointer
    rclcpp::spin(node);
    rclcpp::shutdown();

    return 0;
}

File: src/my_py_pkg/my_py_pkg/smartphone.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import (
    String,
)  # need to add this dependency in package.xml too


class SmartphoneNode(Node):
    def __init__(self):
        super().__init__("smartphone")

        self.subcriber = self.create_subscription(
            String, "robot_news", self.callback_robot_news, 10
        )
        self.get_logger().info("Python Smartphone node has started!")

    def callback_robot_news(self, msg: String):
        self.get_logger().info(msg.data)


def main(args=None):
    rclpy.init(args=args)  # initialize ROS
    node = SmartphoneNode()  # create node
    rclpy.spin(node)  # allows the node to run indefintely
    rclpy.shutdown()  # shutdown node


if __name__ == "__main__":
    main()
